lib/web_ui/dev/screenshot_manager.dart

```diff
@@ -0,0 +1,224 @@
+  /// used by tests (which is inside the browser).
+  int _heightOfFooter;
+
+  /// Used as a suffix for the temporary file names used for screenshots.
+  int _fileNameCounter = 0;
+
+  /// Capture a screenshot of entire simulator.
+  ///
+  /// Example screenshot with dimensions: W x H.
+  ///
+  ///  <----------  W ------------->
+  ///  _____________________________
+  /// | Phone Top bar (clock etc.)  |   Ʌ
+  /// |_____________________________|   |
+  /// | Broswer search bar          |   |
+  /// |_____________________________|   |
+  /// | Web page content            |   |
+  /// |                             |   |
+  /// |                             |
+  /// |                             |   H
+  /// |                             |
+  /// |                             |   |
+  /// |                             |   |
+  /// |                             |   |
+  /// |                             |   |
+  /// |_____________________________|   |
+  /// | Phone footer bar            |   |
+  /// |_____________________________|   V
+  ///
+  /// After taking the screenshot top and bottom parts of the image by
+  /// [_heightOfHeader] and [_heightOfFooter] consecutively. Hence web content
+  /// has the dimensions:
+  ///
+  /// W x (H - [_heightOfHeader] - [_heightOfFooter])
+  ///
+  /// [region] is used to decide which part of the web content will be used in
+  /// test image. It includes starting coordinate x,y as well as height and
+  /// width of the area to capture.
+  ///
+  /// Uses simulator tool `xcrun simctl`'s 'screenshot' command.
+  Future<Image> capture(Rectangle region) async {
+    final String filename = 'screenshot${_fileNameCounter}.png';
+    _fileNameCounter++;
+
+    await IosSafariArgParser.instance.iosSimulator.takeScreenshot(
+        filename, environment.webUiSimulatorScreenshotsDirectory);
+
+    final io.File file = io.File(path.join(
+        environment.webUiSimulatorScreenshotsDirectory.path, filename));
+    List<int> imageBytes;
+    if (!file.existsSync()) {
+      throw Exception('Failed to read the screenshot '
+          'screenshot${_fileNameCounter}.png.');
+    }
+    imageBytes = await file.readAsBytes();
+
+    final Image screenshot = decodePng(imageBytes);
+    // Image with no footer and header.
+    final Image content = copyCrop(
+      screenshot,
+      0,
+      _heightOfHeader,
+      screenshot.width,
+      screenshot.height - _heightOfFooter - _heightOfHeader,
+    );
+
+    return (region == null)
+        ? content
+        : copyCrop(
+            content,
+            region.left.toInt(),
+            region.top.toInt(),
+            region.width.toInt(),
+            region.height.toInt(),
+          );
+  }
+}
+
+const String _kBrowserChrome = 'chrome';
+const String _kBrowserIOSSafari = 'ios-safari';
+
+typedef ScreenshotManagerFactory = ScreenshotManager Function();
+
+/// Abstract class for taking screenshots in different browsers.
```

Comment:
nit: s/in different browsers/in one of the browsers/ (the former implies that the same class takes screenshots in different browsers, while in reality each class works with one of the browsers)

Reply:
done